### ISSUE TITLE
Tagging assertions with an ID

### ISSUE THREAD

**Issue by zhassan-aws** [diffblue/cbmc#6692]
Is there a way to tag assertions in the Goto programs with an ID? In [Kani](https://github.com/model-checking/kani), we're trying to find a way to tag checks in the Goto program that Kani generates with a unique ID so that we can refer to them when post-processing CBMC results. The only way I found is to add them to the assert message, e.g.:
```
[foo.assertion.2] line 4 [KANI_CHECK_ID_reach.ec777b2e::reach_0] assertion failed: x < 4: FAILURE
[foo.assertion.4] line 6 [KANI_CHECK_ID_reach.ec777b2e::reach_1] assertion failed: x == 2: SUCCESS
[foo.assertion.6] line 9 [KANI_CHECK_ID_reach.ec777b2e::reach_2] assertion failed: x <= 5: SUCCESS
```
that is then stripped off during post-processing. This is pretty ugly though.

**Comment by tautschnig** [comment]
Assertions have a unique id, which is what you see in square brackets at the beginning. This id, however, is automatically generated based on the property class. So you'd have the option of altering the property class, but I'm not entirely sure that's the best option for you.

I'm thinking the approach you showed might even be the best one?

**Comment by tautschnig** [comment]
I should add that if we got #6629 merged then it would be possible to set the id (provided it genuinely is unique) upfront, which perhaps is the solution you're looking for?

**Comment by zhassan-aws** [comment]
Yes, #6629 should do it.

**Comment by tautschnig** [comment]
As I have closed #6629 for it increases complexity: the best approach for code generating assertions (in Kani) would be:
1) generate an `assert` statement, not a function call. (Perhaps Kani already does this.)
2) Set the `property_class` in the source location of that `assert` statement.
The resulting ID will have a prefix (the function that the `assert` statement lives in) and a suffix (to ensure unique identifiers), but should maintain the `property_class` as part of the id.
